internal/checker/types.go

```diff
@@ -1094,6 +1109,38 @@ type Signature struct {
 	composite                *CompositeSignature
 }
 
+func (s *Signature) TypeParameters() []*Type {
+	if len(s.typeParameters) == 0 {
```

Comment:
I think here and below, this can also just be `return s.typeParameters`. `len(nil)` just returns 0, so typically there's no need to distinguish between a `nil` and an empty slice.
Then the code in signature help can just do `if len(signature.TypeParameters() != 0 ...`.